src/API.jl

```diff
@@ -1161,25 +1175,46 @@ function precompile(ctx::Context; internal_call::Bool=false, kwargs...)
                 end
             end
         end
+        push!(tasks, task)
+    end
+    while !interrupted && any(!istaskdone, tasks)
+        try
+            sleep(0.001)
```

Comment:
This thing looks pretty weird. Can't this somehow be done in a blocking way instead of spinning on a `sleep`?

Reply:
There may be something tidier but I couldn't figure one out.
We can't use `@sync` as we want to exit before all the compilecache tasks are done when a `^C` is thrown.
And we can't do a single `wait()` here because we need to check for both `!interrupted && any(!istaskdone, tasks)` and not block the other async tasks

Reply:
Figured it out. Now updated